model_interfaces/interface.py

```diff
@@ -27,6 +31,14 @@ def message_to_agent(self, user_message: str) -> Tuple[str, datetime, datetime]:
     def name(self):
         return self.__class__.__name__
 
+    @property
+    def save_path(self):
+        return PERSISTENCE_DIR.joinpath(f"{self.dataset_name} - {self.name}")
+
+    @property
+    def save_name(self):
+        return f"{self.dataset_name} - {self.name}"
```

Comment:
This doesn't seem right. The unique agent name should be `<run_name> - <self.name>`. Also, `save_name` can be used in `save_path`.